commons-macros/src/main/scala/com/avsystem/commons/macros/serialization/GenCodecMacros.scala

```diff
@@ -87,15 +86,15 @@ class GenCodecMacros(ctx: blackbox.Context) extends CodecMacroCommons(ctx) with
       val depNames = depNamesMap(generated.map(_._1))
 
       def depDeclaration(sym: Symbol, depTpe: Type) =
-        q"lazy val ${depNames(sym)} = ${dependency(depTpe, tcTpe, s"for generated member ${sym.name}")}"
+        if (isScalaJs) q"def ${depNames(sym)} = ${dependency(depTpe, tcTpe, s"for generated member ${sym.name}")}"
```

Comment:
I wonder how would that work with DCE if we made a "manual", unsynchronized lazy vals, e.g.

```scala
def depDeclaration(sym: Symbol, depTpe: Type) = {
  val depName = depNames(sym)
  val depVarName = TermName("_" + depName.toString)
  q"""
    private[this] var $depVarName: $depTpe = null
    def $depName = {
      if($depVarName == null) {
        $depVarName = ${dependency(depTpe, tcTpe, s"for generated member ${sym.name}")}
      }
      $depVarName
    }
  """
}
```

This version could be used by both JVM and JS.

Reply:
It generates similar amount of JS code. Simple defs reduce it significantly.

Reply:
What about `def`s in both cases? 

Defs:
```
[info] Benchmark                                                       Mode  Cnt        Score       Error  Units
[info] c.a.c.core.GenCodecBenchmarks.testCaseClassCodec               thrpt   20  2936094.002 â–’ 49606.878  ops/s
[info] c.a.c.rpc.akka.serialization.GenCodecBenchmark.conversionTest  thrpt   20   169360.706 â–’  2359.861  ops/s
[info] c.a.c.rpc.akka.serialization.GenCodecBenchmark.readTest        thrpt   20   316837.049 â–’  3652.074  ops/s
[info] c.a.c.rpc.akka.serialization.GenCodecBenchmark.writeTest       thrpt   20   552318.464 â–’  6536.033  ops/s
```

Lazy vals:
```
[info] Benchmark                                                       Mode  Cnt        Score       Error  Units
[info] c.a.c.core.GenCodecBenchmarks.testCaseClassCodec               thrpt   20  3425750.118 â–’ 48293.942  ops/s
[info] c.a.c.rpc.akka.serialization.GenCodecBenchmark.conversionTest  thrpt   20   171961.270 â–’  2343.882  ops/s
[info] c.a.c.rpc.akka.serialization.GenCodecBenchmark.readTest        thrpt   20   323721.652 â–’  4722.604  ops/s
[info] c.a.c.rpc.akka.serialization.GenCodecBenchmark.writeTest       thrpt   20   490266.148 â–’ 24286.542  ops/s
```